Switch to Maplibre GL, Style the basemap, Add Ubuntu Mono fonts

ireo00 opened the pull request.

ireo00 committed:
Switch to Maplibre GL, Style the basemap, Add Ubuntu Mono fonts

jamaps committed:
add maplibre and pmtiles to package.json

jamaps reviewed (changes requested):
The base map looks great! even better than before!

I had to add `maplibre-gl` and `pmtiles` to the `package.json` and then run `npm install` to get it working

A couple things to fix before merging
- for the `toronto-building-permits` page, the layers/buttons aren't loading/working properly, I'm not sure why. It is also missing the range slider for selecting by year
- the `toronto-building-permits` page is also missing the subway and municipal boundary layers
- on both pages, the font size for the items below the map have changed,  and there is overflow on some of the labels. The fonts for the buttons should be the same as what we have on the live site - https://schoolofcities.github.io/gentle-density/toronto-building-permits

e.g. comparing your branch with the live site shows the difference

![Screenshot from 2024-02-09 14-35-09](https://github.com/schoolofcities/gentle-density/assets/11054490/c932aaef-f0e0-48a1-8edf-e5f65989431f)
![Screenshot from 2024-02-09 14-34-40](https://github.com/schoolofcities/gentle-density/assets/11054490/b21e1da4-986c-4bcd-8307-881a558e6979)

ireo00 committed:
fix 'toronto-building-permits' map

- subway and municipal boundary layers are now visible
- adjusted the overflow of some labels
- fixed all layers/buttons, files are loading properly
	- 'x-secondary' & 'x-rearyard' icons need extra coding to work (specify the image source, then use the addImage function to load it on to the map)

ireo00 committed:
Update MapTorontoSecondaryLaneway.svelte

ireo00 committed:
fix the order of layers, edit map zoom

jamaps committed:
bbox adjustment and font size change

jamaps committed:
build

jamaps committed:
trying to make maps centre properly in bbox

jamaps committed:
build
